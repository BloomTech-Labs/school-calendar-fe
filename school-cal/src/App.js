import React from 'react';
import logo from './logo.svg';
import './App.css';
import Login from './components/logincomp'; 
import {BrowserRouter as Router , Route} from 'react-router-dom'
import {Admin} from './components/DummyAdminDashboard'
import privateRoute from './routes/privateRoute.js'


function App() {
  return ( 
    <Router>
    <div className="App">
    <privateRoute exact path = "/admin" component ={Admin}/>  
    <Route exact path = "/login" component = {Login}/> 
    </div> 
    </Router>
  );
}

export default App;
