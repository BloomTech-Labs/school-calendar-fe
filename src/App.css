.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.navbar {
  align-items: center;
  background:#75b79e;
  display: flex;
  height: 70px;
  justify-content: space-around;
  margin: 10px;
  padding: 0 3%;
  width: 100%;
  -webkit-box-shadow: 0px 2px 15px -8px rgba(0, 0, 0, 0.42);
  -moz-box-shadow: 0px 2px 15px -8px rgba(0, 0, 0, 0.42);
  box-shadow: 0px 2px 15px -8px rgba(0, 0, 0, 0.42);
}