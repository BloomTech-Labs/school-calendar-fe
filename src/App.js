import React from 'react';
import { Route } from 'react-router-dom';

import Splash from './components/Splash/';
import Home from './components/Home/';
// import PrivateRoute from "./components/PrivateRoute";
import Redirect from './components/Redirect';
import Events from './components/Home/Events';


import './App.css';

function App() {
	return (
		<div className='App'>
			<Route exact path='/'>
				<Splash />
			</Route>
			{/* Remember to change back to private */}
			{/* <PrivateRoute path="/home">
        <Home />
      </PrivateRoute> */}
			
			<Route path='/events'>
				<Events />
			</Route>
			<Route path='/redirect'>
				<Redirect />
			</Route>
			<Route path='/home'>
				<Home />
			</Route>
		</div>
	);
}

export default App;
