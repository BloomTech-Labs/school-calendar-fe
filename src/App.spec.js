import React from 'react';
import { shallow } from 'enzyme';
import App, {initializeAnalytics} from './App';
import Welcome from './components/Welcome'

// Grab the module that we want to mock part of
var auth = require('./contexts/auth');

// Replace the real useAuth() function with a mock function we can control
auth.useAuth = jest.fn(
  () => {
    console.log("*** Mock useAuth **")

    return { googleApi: jest.fn() }
  });

describe('Initializing Components', () => {
  it('should render initializeAnalytics', () => {
    const component = shallow(<initializeAnalytics />);

    expect(component).toMatchSnapshot();
  });

  it('should render App correctly', () => {
    const component = shallow(<App />);

    expect(component).toMatchSnapshot();
  });

  it('should render Welcome correctly', () => {
    const welcome = render(<Welcome />);
    let signin = <span className="signin">Sign in with Google</span>

    expect(welcome.contains(signin)).toEqual(true);
   
  });


});