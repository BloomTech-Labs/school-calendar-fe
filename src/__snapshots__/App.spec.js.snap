// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initializeAnalytics should render correctly 1`] = `<initializeAnalytics />`;

exports[`initializeAnalytics should render correctly 2`] = `
<div
  className="App"
>
  <Route
    exact={true}
    path="/"
  >
    <Splash />
  </Route>
  <Route
    path="/authenticate/google"
  >
    <Authenticate />
  </Route>
  <Route
    path="/home"
  >
    <Home />
  </Route>
  <Route
    path="/events"
  >
    <Events />
  </Route>
</div>
`;
